src/main/java/hudson/plugins/ec2/EC2SpotSlave.java

```diff
@@ -94,17 +99,15 @@ public void terminate() {
                                     */
                                     LOGGER.info("EC2 instance already terminated: " + instanceId);
                                 } else {
+                                    // check to see if there is a dynamic keypair associated with this instance, and if so, clean it up
+                                    cleanupSshKeyPairIfNeeded();
+                                    // send request to terminate instance
                                     TerminateInstancesRequest request = new TerminateInstancesRequest(Collections.singletonList(instanceId));
-                                    try {
-                                        ec2.terminateInstances(request);
-                                        LOGGER.info("Terminated EC2 instance (terminated): " + instanceId);
-                                    } catch (AmazonClientException e) {
-                                        // Spot request is no longer valid
-                                        LOGGER.log(Level.WARNING, "Failed to terminate the Spot instance: " + instanceId, e);
-                                    }
+                                    ec2.terminateInstances(request);
+                                    LOGGER.info("Terminated EC2 instance (terminated): " + instanceId);
```

Comment:
So if `cleanupSshKeyPairIfNeeded` fails, then instance will not be deleted. I think you should revert the error-handling changes in this block and wrap `cleanupSshKeyPairIfNeeded` in its own `try`-`catch` just to be sure.

Reply:
i've reverted the error handling and wrapped the `cleanupSshKeyPairIfNeeded()` call with a `try/catch/log`